var/spack/repos/builtin/packages/php/package.py

```diff
@@ -68,3 +88,8 @@ def patch(self):
     def autoreconf(self, spec, prefix):
         bash = which("bash")
         bash("./buildconf", "--force")
+
+    @when("@8:")
+    def configure_args(self):
+        args = [f"--with-libiconv={self.spec['libiconv'].prefix}"]
```

Comment:
Because this triggers `configure: WARNING: unrecognized options: --with-libiconv`, I assume this is supposed to be `--with-libiconv=`.
```suggestion
        args = [f"--with-iconv={self.spec['libiconv'].prefix}"]
```
This fixes the build of `php@8` on Ubuntu 22.04 for me.

Supporting the new php@8: in this way is also what https://github.com/php/php-src/issues/12213#issuecomment-1826979238 appears to suggest.